lnrpc/invoicerpc/invoice.proto

```diff
@@ -244,28 +352,34 @@ message ListInvoiceResponse {
 message InvoiceSubscription {
     /**
     If specified (non-zero), then we'll first start by sending out
-    notifications for all added indexes with an add_index greater than this
-    value. This allows callers to catch up on any events they missed while they
+    all invoices that have an last_update_index greater than this value. This allows
+    callers to retrieve up to date versions of invoices that changed when they
     weren't connected to the streaming RPC.
     */
-    uint64 add_index = 1;
```

Comment:
With `update_index` they can do exactly do same, because the creation is an event as well?

Reply:
I was thinking they would be separate for backwards compatibility. We can have a migration that copies over the existing settle_index entries into the update index, and client indices would be compatible.

If adds are also in the update index, they must have an index that precedes the settle index which breaks old values.

May this isn't so bad, and people just reindex?

Reply:
So the migration would be to create update records as follows?
```
update index: settle index
invoice_number: add index
state: settled
settle_date: settle_date
amt_paid: amt_paid
```

I think it breaks an assumption anyway in that from the migration onwards not every value of update_index will be a settle event anymore.

What I don't like about two indices is that you need to call SubscribeInvoices twice to get a complete picture of the current state. 

I think I am in favor of just leaving settle and add index what they are - we need to do that anyway because we are only adding a subserver - and add an updateindex to the db that includes the created event as well. For the migration, we can first insert all the existing adds in the updateindex and then all existing settles.